File: MyVoiceTests/LanguagePreferenceTests.swift
import Testing
import Foundation
@testable import MyVoice

struct LanguagePreferenceTests {
    private func freshDefaults(_ suite: String = UUID().uuidString) -> UserDefaults {
        let d = UserDefaults(suiteName: suite)!
        d.removePersistentDomain(forName: suite)
        return d
    }

    @Test func defaultIsAutoWhenKeyAbsent() {
        let defaults = freshDefaults()
        #expect(LanguagePreference.load(from: defaults) == .auto)
    }

    @Test func roundTripPersistence() {
        let defaults = freshDefaults()
        LanguagePreference.english.save(to: defaults)
        #expect(LanguagePreference.load(from: defaults) == .english)

        LanguagePreference.turkish.save(to: defaults)
        #expect(LanguagePreference.load(from: defaults) == .turkish)

        LanguagePreference.auto.save(to: defaults)
        #expect(LanguagePreference.load(from: defaults) == .auto)
    }

    @Test func invalidRawValueFallsBackToAuto() {
        let defaults = freshDefaults()
        defaults.set("klingon", forKey: LanguagePreference.userDefaultsKey)
        #expect(LanguagePreference.load(from: defaults) == .auto)
    }

    @Test func rawValuesMatchWhisperCodes() {
        #expect(LanguagePreference.auto.rawValue == "auto")
        #expect(LanguagePreference.english.rawValue == "en")
        #expect(LanguagePreference.turkish.rawValue == "tr")
    }

    @Test func displayNamesAreHumanReadable() {
        #expect(LanguagePreference.auto.displayName == "Auto-detect")
        #expect(LanguagePreference.english.displayName == "English")
        #expect(LanguagePreference.turkish.displayName == "Türkçe")
    }

    @Test func allCasesCoversThreeOptions() {
        #expect(LanguagePreference.allCases.count == 3)
        #expect(LanguagePreference.allCases.contains(.auto))
        #expect(LanguagePreference.allCases.contains(.english))
        #expect(LanguagePreference.allCases.contains(.turkish))
    }
}

File: windows/tests/MyVoice.Windows.Tests/LanguagePreferenceTests.cs
using MyVoice.Windows.Core;

namespace MyVoice.Windows.Tests;

// Ported one-to-one from MyVoiceTests/LanguagePreferenceTests.swift; persistence goes through settings.json.
public sealed class LanguagePreferenceTests : IDisposable
{
    readonly string _path = Path.Combine(Path.GetTempPath(), $"myvoice-language-{Guid.NewGuid()}.json");

    public void Dispose() => File.Delete(_path);

    [Fact]
    public void DefaultIsAutoWhenKeyAbsent()
    {
        File.WriteAllText(_path, """{"debug": false}""");
        Assert.Equal(LanguagePreference.Auto, LanguagePreference.FromCode(AppSettings.Load(_path, out _).Language));
        Assert.Equal(LanguagePreference.Auto, LanguagePreference.FromCode(null));
    }

    [Fact]
    public void RoundTripPersistence()
    {
        foreach (var preference in new[] { LanguagePreference.English, LanguagePreference.Turkish, LanguagePreference.Auto })
        {
            new AppSettings { Language = preference.Code }.Save(_path);
            Assert.Equal(preference, LanguagePreference.FromCode(AppSettings.Load(_path, out _).Language));
        }
    }

    [Fact]
    public void InvalidRawValueFallsBackToAuto() => Assert.Equal(LanguagePreference.Auto, LanguagePreference.FromCode("klingon"));

    [Fact]
    public void RawValuesMatchWhisperCodes()
    {
        Assert.Equal("auto", LanguagePreference.Auto.Code);
        Assert.Equal("en", LanguagePreference.English.Code);
        Assert.Equal("tr", LanguagePreference.Turkish.Code);
    }

    [Fact]
    public void DisplayNamesAreHumanReadable()
    {
        Assert.Equal("Auto-detect", LanguagePreference.Auto.DisplayName);
        Assert.Equal("English", LanguagePreference.English.DisplayName);
        Assert.Equal("Türkçe", LanguagePreference.Turkish.DisplayName);
    }

    [Fact]
    public void AllCasesCoversThreeOptions()
    {
        Assert.Equal(3, LanguagePreference.All.Count);
        Assert.Contains(LanguagePreference.Auto, LanguagePreference.All);
        Assert.Contains(LanguagePreference.English, LanguagePreference.All);
        Assert.Contains(LanguagePreference.Turkish, LanguagePreference.All);
    }
}
